neofetch

```diff
@@ -4199,11 +4199,16 @@ cache_uname() {
     kernel_machine="${uname[2]}"
 
     if [[ "$kernel_name" == "Darwin" ]]; then
-        IFS=$'\n' read -d "" -ra sw_vers < <(awk -F'<|>' '/string/ {print $3}' \
+        IFS=$'\n' read -d "" -ra sw_vers < <(awk -F'<|>' '/key|string/ {print $3}' \
                             "/System/Library/CoreServices/SystemVersion.plist")
-        darwin_name="${sw_vers[2]}"
-        osx_version="${sw_vers[3]}"
-        osx_build="${sw_vers[0]}"
+        for i in "${!sw_vers[@]}"; do
```

Comment:
Does this work?

```sh
for ((i=0;i<${#sw_vers[@]};i++)) {
    case ${sw_vers[i]} in
        ProductName)          darwin_name=${sw_vers[i+1]} ;;
        ProductVersion)       osx_version=${sw_vers[i+1]} ;;
        ProductBuildVersion)  osx_build=${sw_vers[i+1]}   ;;
    esac
}
```

Reply:
Yes, it works!
We might also increase the step size to 2, since starting with the first, every second value should be a key, if `SystemVersion.plist` is correctly formattted.
From my `SystemVersion.plist`:
```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
        <key>BuildID</key>
        <string>*removed*</string>
        <key>ProductBuildVersion</key>
        <string>15E216</string>
        <key>ProductCopyright</key>
        <string>1983-2018 Apple Inc.</string>
        <key>ProductName</key>
        <string>iPhone OS</string>
        <key>ProductVersion</key>
        <string>11.3</string>
        <key>SystemImageID</key>
        <string>*removed*</string>
</dict>
</plist>
```

What's your opinion on this?